docs/content/test/test.md

```diff
@@ -236,40 +271,133 @@ An update test is a test that creates the target resource and then makes updates
+- `BootstrapKMSKey(t *testing.T)`
+  - Uses `global` location, `ENCRYPT_DECRYPT` for the purpose, and the corresponding key name for that purpose. 
 
+Example usage:
+
+{{< tabs "bootstrap-cryptokeys" >}}
+{{< tab "MMv1" >}}
+```yaml
+examples:
+  - name: service_resource_basic
+    primary_resource_id: example
+    vars:
+      kms_key_name: 'kms-key'
+    test_vars_overrides:
+      kms_key_name: 'acctest.BootstrapKMSKey(t).CryptoKey.Name'
+```
+{{< /tab >}}
+{{< tab "Handwritten" >}}
 ```go
-func TestSignatureAlgorithmDiffSuppress(t *testing.T) {
-   cases := map[string]struct {
-      Old, New           string
-      ExpectDiffSuppress bool
-   }{
-      "ECDSA_P256 equivalent": {
-         Old:                "ECDSA_P256_SHA256",
-         New:                "EC_SIGN_P256_SHA256",
-         ExpectDiffSuppress: true,
-      },
-      // Additional cases excluded for brevity
+func TestAccProductResource_update(t *testing.T) {
+   t.Parallel()
+
+   context := map[string]interface{}{
+      "kms": acctest.BootstrapKMSKey(t).CryptoKey.Name,
+      // other variables
    }
+   // rest of test
+}
+```
+{{< /tab >}}
+{{< /tabs >}}
 
-   for tn, tc := range cases {
-      if signatureAlgorithmDiffSuppress("signature_algorithm", tc.Old, tc.New, nil) != tc.ExpectDiffSuppress {
-         t.Errorf("bad: %s, %q => %q expect DiffSuppress to return %t", tn, tc.Old, tc.New, tc.ExpectDiffSuppress)
-      }
+### IAM resources
+
+Specify member/role pairs that should always exist in the default test project. `{project_number}` will be replaced with the default project's project number.
+
+Permissions attached to resources created _in_ a test should instead be provisioned with standard terraform resources.
+
+Example usage:
+
+{{< tabs "bootstrap-iam" >}}
+{{< tab "MMv1" >}}
+```yaml
+examples:
+  - name: service_resource_basic
+    primary_resource_id: example
+    bootstrap_iam:
+      - member: "serviceAccount:service-{project_number}@gcp-sa-healthcare.iam.gserviceaccount.com"
+        role: "roles/bigquery.dataEditor"
+```
+{{< /tab >}}
+{{< tab "Handwritten" >}}
+```go
+func TestAccProductResource_update(t *testing.T) {
+   t.Parallel()
+
+   acctest.BootstrapIamMembers(t, []acctest.IamMember{
+      {
+         Member: "serviceAccount:service-{project_number}@gcp-sa-pubsub.iam.gserviceaccount.com",
+         Role:   "roles/cloudkms.cryptoKeyEncrypterDecrypter",
+      },
+   })
+   // rest of test
+}
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+### Networks
+
+Bootstrapping networks can be useful for two reasons:
+
+1. Resources like service_networking_connection use a consumer network and create a complementing tenant network which we don't control. These tenant networks never get cleaned up and they can accumulate to the point where a limit is reached for the organization. By reusing a consumer network across test runs, we can reduce the number of tenant networks that are needed. (Googlers: See b/146351146 for more context.)
```

Comment:
```suggestion
1. Resources like `google_service_networking_connection` use a consumer network and create a complementing tenant network which we don't control. These tenant networks never get cleaned up and they can accumulate to the point where a limit is reached for the organization. By reusing a consumer network across test runs, we can reduce the number of tenant networks that are needed. (Googlers: See b/146351146 for more context.)
```
Not sure if we want to use the Terraform resource name here or just point it as a general reference

Reply:
I think we should list explicit resource names since people reading this document won't necessarily know if a particular API resource has this behavior. If we find additional similar resources in the future we can add them here as well.